src/reducers/measurements/types.ts

```diff
@@ -0,0 +1,100 @@
+import { measurementIdSymbol } from "../../util/globals";
+
+// -- Measurements JSON types -- //
+
+/**
+ * Measurements are allowed to have arbitrary metadata.
+ * Matching types allowed in Augur's measurements schema
+ * <https://github.com/nextstrain/augur/blob/3f72c40897a80132099729d5d00a6718e76e0e9e/augur/data/schema-measurements.json#L152>
+ */
+type MeasurementMetadata = string | number | boolean
+
+export const measurementsDisplayValues = ["raw", "mean"];
+export type MeasurementsDisplay = (typeof measurementsDisplayValues)[number]
+export const isMeasurementsDisplay = (x: any): x is MeasurementsDisplay => measurementsDisplayValues.includes(x);
+
+interface JsonMeasurement {
+  readonly strain: string
+  readonly value: number
+  readonly [key: string]: MeasurementMetadata
+}
+
+interface JsonCollectionDisplayDefaults {
+  readonly group_by?: string
+  readonly measurements_display?: MeasurementsDisplay
+  readonly show_overall_mean?: boolean
+  readonly show_threshold?: boolean
+}
+
+interface JsonCollectionField {
+  readonly key: string
+  readonly title?: string
+}
+
+interface JsonCollectionGrouping {
+  readonly key: string
+  readonly order?: MeasurementMetadata[]
+}
+
+export interface JsonCollection {
+  readonly display_defaults?: JsonCollectionDisplayDefaults
+  readonly fields?: JsonCollectionField[]
+  readonly filters?: string[]
+  readonly groupings: JsonCollectionGrouping[]
+  readonly key: string
+  readonly measurements: JsonMeasurement[]
+  readonly threshold?: number
+  readonly thresholds?: number[]
+  readonly title?: string
+  readonly x_axis_label: string
+}
+
+export interface MeasurementsJson {
+  readonly collections: JsonCollection[]
+  readonly default_collection?: string
+}
+
+// -- Measurements state types -- //
+
+export interface Measurement {
+  [measurementIdSymbol]: number
+  strain: string
+  value: number
+  [key: string]: string | number
+}
+
+export function isMeasurement(x: any): x is Measurement {
+  return Boolean(x[measurementIdSymbol] !== undefined && x.strain && x.value !== undefined)
+}
```

Comment:
_related to https://github.com/nextstrain/nextstrain.org/pull/1073#discussion_r1847218126_

This is prone to bugs, e.g. if we add a required property to `Measurement` and forget to add it in the condition here.

Something like this should be more robust:

```ts
export function asMeasurement(x: Partial<Measurement>): Measurement {
  if (x[measurementIdSymbol] !== undefined && x.strain && x.value !== undefined) {
    return {
       ...x,
       [measurementIdSymbol]: x[measurementIdSymbol],
       strain: x.strain,
       value: x.value,
    }
  }
  throw new Error("Measurement is partial.");
}

export function isMeasurement(x: Partial<Measurement>): x is Measurement {
  try {
    asMeasurement(x);
    return true;
  } catch {
    return false;
  }
}
```

Reply:
Thanks! I missed the later part of that comment thread. Will update.

Reply:
Updated in https://github.com/nextstrain/auspice/pull/1910/commits/0093b4b3addf066682fe2b857840a1c31af3689c